### PR TITLE
feature/SOF-3956

### PR DESCRIPTION
**Pull request by mohammadidinani**

### CODE REVIEW

File: lang/en/docs/software-directory/scripting/jupyter-lab/overview.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Jupyter Lab
+
+
+JupyterLab [^1] is the next-generation user interface for Project Jupyter [^2] offering all the familiar building blocks of the classic Jupyter Notebook [^3] in a flexible and powerful user interface. JupyterLab will eventually replace the classic Jupyter Notebook.
+
+We include support for **Jupyter Lab** within our platform, using interpreter version 2.7.5 for Python programming language.
```
Comment (timurbazhirov) [inline lang/en/docs/software-directory/scripting/jupyter-lab/overview.md]: 1. the operating system default interpreter version 2.7.5 for Python
2. remove `programming language`

File: lang/en/docs/tutorials/other/jupyter.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
+# Jupyter Notebook
```
Comment (timurbazhirov) [inline lang/en/docs/tutorials/other/jupyter.md]: 1. Let's follow the video step by step here too
2. Add a link to exabyte-api-examples repository

File: lang/en/docs/tutorials/other/jupyter.json [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+            "endTime": "0:03:38.000",
+            "startTime": "0:03:35.000"
+        },
+        {
+            "text": "Close the notebook unit and navigates to files tab",
+            "endTime": "0:03:42.000",
+            "startTime": "0:03:39.000"
+        },
+        {
+            "text": "A copy of modified I Python notebooks are saved inside checkpoints folder at the time they are saved",
```
Comment (timurbazhirov) [inline lang/en/docs/tutorials/other/jupyter.json]: Copies of modified

File: lang/en/docs/software-directory/scripting/jupyter-lab/overview.md (hunk 1 above)
Comment (mohammadidinani) [reply lang/en/docs/software-directory/scripting/jupyter-lab/overview.md]: Done.

File: lang/en/docs/tutorials/other/jupyter.json (hunk 3 above)
Comment (mohammadidinani) [reply lang/en/docs/tutorials/other/jupyter.json]: Done.

File: lang/en/docs/tutorials/other/jupyter.md (hunk 2 above)
Comment (mohammadidinani) [reply lang/en/docs/tutorials/other/jupyter.md]: Done.

File: lang/en/docs/software-directory/scripting/jupyter-lab/data.md [hunk 4]
```diff
@@ -0,0 +1,11 @@
+# Jupyter Lab: Structured Representation
```
Comment (timurbazhirov) [inline lang/en/docs/software-directory/scripting/jupyter-lab/data.md]: We need to add a note about accessing the ipynb files and the rest of the files:

```
## Files/storage convention

1. Initially, the root of the Dropbox folder is passed to the application on the start, so the files at the root of the Dropbox folder (link) can be accessed
2. Upon each "Save and Checkpoint" action invoked inside the notebook, the ipynb file is overwritten. A new version is stored in the file system, and a checkpoint is saved to the job inside its directory (link).
3. All notebooks have access to the filesystem accessible to the user on the corresponding computational node, namely the home (link) and share (share) directories. For example, the following command will list the shared directory for the account "exabyte-io", when invoked inside the Jupyter Notebook running on "cluster-007": `<COMMAND HERE>`
```

File: lang/en/docs/tutorials/other/jupyter.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+This tutorial page explains how to create a Jupyter Notebook environment through [Jupyter Lab](../../software-directory/scripting/jupyter-lab/overview.md) application following the below steps.
+
+## Generate RESTFul API Tokens
+
+The Jupyter notebook environment in the present tutorial is used to run an IPython notebook from [Exabyte API Examples Repository](../../rest-api/api-examples.md) in which a connection is made to the RESTFul API to retrieve a list of materials. In order to establish the connection, one should generate RESTFul API tokens following the steps described in [here](../../rest-api/authentication.md). 
+
+## Upload IPython Notebooks
+
+Jupyter Notebook is started on the account [Dropbox](../../data-in-objectstorage/dropbox.md) directory. This directory provides users with access to previously uploaded/created IPython notebooks. Here, **settings.py** file contains the variables required to configure the RESTFul API endpoints and **get_materials_by_formula.ipynb** from the [Exabyte API Examples Github Repository](../../rest-api/api-examples.md) are uploaded to Dropbox to be later used inside the Jupyter notebook environment.
```
Comment (timurbazhirov) [inline lang/en/docs/tutorials/other/jupyter.md]: Add a link to the explanation about the filesystem conventions above

File: lang/en/docs/tutorials/other/jupyter.md [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    Do not use the URL printed in the output file as the notebooks can not be accessed via printed URL for security reasons.
+
+## Save Jupyter Notebooks
+
+**Make sure to save and checkpoint the notebook after introducing any changes**. The "save and checkpoint" Jupyter action overwrites the original notebook loaded from Dropbox and saves a copy of the notebook inside **checkpoints** directory located in the [job working directory](../../jobs-cli/batch-scripts/directories.md#working-directory). The checkpoints will be later accessible through the [Job Files Explorer](../../data-in-objectstorage/files.md) tab.
+
+## Stop Jupyter Environment
+
+When don editing, the Jupyter Notebook environment can be stopped by either clicking the **Quit** button in Jupyter Notebook or [terminating](../../jobs/actions/terminate.md) the job. In either case, **make sure to save any changes you have made before stopping the notebook as unsaved changes will be lost otherwise**.
+
```
Comment (timurbazhirov) [inline lang/en/docs/tutorials/other/jupyter.md]: Add a step

```
## Accessing the modified files

As explained in the [dedicated section](link) the modified ipynb files, as well as the checkpoints at each save, and the other files associated with the job can be accessed from the Dropbox folder, the job and through command-line.
```

File: lang/en/docs/tutorials/other/jupyter.json [hunk 7]
```diff
@@ -0,0 +1,180 @@
+{
+    "descriptionLinks": [
+        "Jupyter Notebook Environment: https://docs.exabyte.io/tutorials/other/jupyter"
+    ],
+    "description": "In this tutorial, we show how to create a Jupyter Notebook environment using our Web Interface.",
+    "tags": [
+        {
+            "...": "../../metadata/general.json#/tags"
+        }
```
Comment (timurbazhirov) [inline lang/en/docs/tutorials/other/jupyter.json]: Let's add tags for data science and jupyter here too

File: lang/en/docs/software-directory/scripting/jupyter-lab/data.md (hunk 4 above)
Comment (mohammadidinani) [reply lang/en/docs/software-directory/scripting/jupyter-lab/data.md]: Added to Jupyter lab application overview page.

File: lang/en/docs/tutorials/other/jupyter.json (hunk 7 above)
Comment (mohammadidinani) [reply lang/en/docs/tutorials/other/jupyter.json]: Done.

File: lang/en/docs/tutorials/other/jupyter.md (hunk 5 above)
Comment (mohammadidinani) [reply lang/en/docs/tutorials/other/jupyter.md]: `Access Modified Files` section has it.

File: lang/en/docs/tutorials/other/jupyter.md (hunk 6 above)
Comment (mohammadidinani) [reply lang/en/docs/tutorials/other/jupyter.md]: Done.

File: lang/en/docs/tutorials/other/jupyter.json (hunk 7 above)
Comment (timurbazhirov) [reply lang/en/docs/tutorials/other/jupyter.json]: We need separate tags for jupyter and data science. The latter has to go to the `data` folder.

File: lang/en/docs/software-directory/scripting/jupyter-lab/overview.md [hunk 8]
```diff
@@ -0,0 +1,26 @@
+# Jupyter Lab
+
+
+JupyterLab [^1] is the next-generation user interface for Project Jupyter [^2] offering all the familiar building blocks of the classic Jupyter Notebook [^3] in a flexible and powerful user interface. JupyterLab will eventually replace the classic Jupyter Notebook.
+
+We include support for **Jupyter Lab** within our platform, using the operating system default interpreter version 2.7.5 for Python.
+
+## Files/Storage Convention
```
Comment (timurbazhirov) [inline lang/en/docs/software-directory/scripting/jupyter-lab/overview.md]: This should go to a separate section/file, the overview should contain links only. I suggested including it in data earlier. You can create a separate file too.

File: lang/en/docs/tutorials/other/jupyter.md [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+## Access Modified Files
+
+As explained in the [dedicated section](../../software-directory/scripting/jupyter-lab/overview.md) the modified IPython files, as well as the checkpoints at each save, and the other files associated with the job can be accessed from the Dropbox folder, the job and through command-line.
+
+## Animation
+
+We demonstrate the steps mentioned above in the animation below.
+
+<div class="video-wrapper">
+<iframe class="gifffer" width="100%" height="100%" src="https://www.youtube.com/embed/G5L52T6fjeQ" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
```
Comment (timurbazhirov) [inline lang/en/docs/tutorials/other/jupyter.md]: The link is to a band gap calculation

**Pull request merged**